solidity/interface-gen-tool.md

```diff
@@ -0,0 +1,82 @@
+- No pseudo-constructor support: The OP Stack codebase heavily relies on pseudo-constructors for type-safe constructor argument passing, which are not supported by this method.
+- Unnecessary custom error replication: The generated interfaces include custom errors from the contract file, which is not relevant or necessary for our use case and cannot be optionally omitted.
+- Interface Name: The name of the generated interface class is not customizable and is currently hardcoded to be the same as the contract name. This does not allow for a clear distinction between interfaces and contracts.
+
+These limitations make this approach suboptimal for our specific needs in the OP Stack codebase. Developers still have to manually review and update interfaces after they are generated.
+
+## Goals
+
+- Automate the process of creating and updating interfaces for contracts in the OP Stack codebase. This reduces the overhead of manually creating and updating interfaces.
+- Ensure generated interfaces correctly: represent enum types and preserve their named variants, support pseudo-constructors, exclude custom errors from the generated interface files, allow for customizable interface names.
+- Maintain or improve the current efficiency gains in compile times achieved through the use of interfaces. If compile times are increased, the benefits of automatically generated interfaces should still outweigh the additional time.
+- Enhance the developer experience by reducing the overhead of interface management.
+
+# Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+The proposed solution is an interface generation script that will function similarly to `forge inspect` and `cast interface`, but with enhanced capabilities to address our specific needs:
+
+- Support for pseudo-constructors, which are crucial for type-safe constructor argument passing in the OP Stack codebase.
+- Correct representation of enum types, preserving their named variants and ensuring type safety.
+- Exclusion of custom errors from the generated interface files, as they are not required for our use case.
+- Allow for customizable interface names.
+
+## Resource Usage
+
+<!-- What is the resource usage of the proposed solution?
+Does it consume a large amount of computational resources or time? -->
+
+It should be a relatively fast script.
+
+# Risks & Uncertainties
+
+<!-- An overview of what could go wrong.
+Also any open questions that need more work to resolve. -->
+
+- How should Types and Events be handled?
```

Comment:
I think let's not worry about this for now and just redefine the errors and events in the interface.